File: library-javascript/LibraryView/src/app/book/model/book.ts
export class PublishingHouse {
  name: String;
  id: String;

  constructor(id?: String, name?: String) {
    this.id = id || '';
    this.name = name || '';
  }
}

export class Book {
  id: String;
  title: String;
  isbn: String;
  author: String;
  numberOfPages: number;
  genres: String[];
  publishingHouse: PublishingHouse;

  constructor(
    id?: string,
    title?: string,
    isbn?: string,
    author?: string,
    numberOfPages?: number,
    genres?: string[],
    publishingHouse?: PublishingHouse
  ) {
    this.id = id || '';
    this.title = title || '';
    this.isbn = isbn || '';
    this.author = author || '';
    this.numberOfPages = numberOfPages || 0;
    this.genres = genres || [];
    this.publishingHouse = publishingHouse || new PublishingHouse();
  }
}

export class NewBook {
  id: String;
  title: String;
  isbn: String;
  author: String;
  numberOfPages: number;
  genres: String[];
  publishingHouse: String;

  constructor(
    id?: string,
    title?: string,
    isbn?: string,
    author?: string,
    numberOfPages?: number,
    genres?: string[],
    publishingHouse?: String
  ) {
    this.id = id || '';
    this.title = title || '';
    this.isbn = isbn || '';
    this.author = author || '';
    this.numberOfPages = numberOfPages || 0;
    this.genres = genres || [];
    this.publishingHouse = publishingHouse || '';
  }
}

File: library-api-books/src/main/java/pg/eti/book/entity/Book.java
package pg.eti.book.entity;

import jakarta.persistence.*;
import lombok.*;
import lombok.experimental.SuperBuilder;
import pg.eti.literature.entity.Literature;
import pg.eti.publishinghouse.entity.PublishingHouse;

import java.util.List;

@SuperBuilder
@Getter
@Setter
@NoArgsConstructor
@AllArgsConstructor
@EqualsAndHashCode(callSuper = true)
@Entity
@ToString(callSuper = true)
@Table(name = "books")
public class Book extends Literature {

	private String author;

	@Column(unique = true)
	private String isbn;

	private List<String> genres;

	@ManyToOne
	@JoinColumn(name = "publishing_house_id")
	private PublishingHouse publishingHouse;
}
